### PR TITLE
Enable assertions and severity markup for fuzz targets

### PR DESCRIPTION
**Pull request by fmeum**
Allows using `assert` in fuzz targets combined with a custom exception type for security bugs (with an optionally specified severity).

### CODE REVIEW

File: agent/src/main/java/com/code_intelligence/jazzer/api/Severity.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * The severity level of a detected security issue.
+ */
+public enum Severity {
+  /**
+   * Examples of security issues of critical severity include: remote code execution
+   */
+  CRITICAL,
+  /**
+   * Examples of security issues of high severity include: SQL injections
```
Comment (fmeum) [inline agent/src/main/java/com/code_intelligence/jazzer/api/Severity.java:26]: These could use some more examples (as well as consensus on whether these rankings are indeed sensible). @sebastianpoeplau Do you have some ideas on what kind of issues to add here that apply to memory-safe languages?

Comment (simonresch) [reply agent/src/main/java/com/code_intelligence/jazzer/api/Severity.java:26]: We could take some items from the OWASP Top 10 list, no? E.g. "Broken Authentication", "Sensitive Data Exposure" could be checked with specialized fuzz targets.

**Review by simonresch — APPROVED** [review]
Cool addition. Changes look good to me.

File: agent/src/main/java/com/code_intelligence/jazzer/api/Severity.java (hunk 1 above)
Comment (fmeum) [reply agent/src/main/java/com/code_intelligence/jazzer/api/Severity.java]: I think that the severity of such rather generic issues would depend a lot more on context than for something like RCE. I have weakened the wording a bit.

**Pull request merged**